### PR TITLE
Upgrade to Jest 29

### PR DESCRIPTION
**Pull request by lachlanjc**
Though development is done on upgrading, we need to wait for the [new version of @emotion/jest](https://github.com/emotion-js/emotion/pull/2885), since it [adds compatibility](https://github.com/emotion-js/emotion/pull/2884) with Jest 29's types, & I'd rather not merge the addition of `strict-peer-dependencies=false` to `.npmrc` needed to make this work right now.

In the meantime, I've closed all the Dependabot PRs since this upgrade requires code changes.

### CODE REVIEW

**Comment by lachlanjc** [comment]
@hasparus any quick tips on why https://github.com/system-ui/theme-ui/actions/runs/3063545168/jobs/4945735744#step:7:34 is failing? This was happening in dev; everything worked if I `ts-expect-error`'d that line, but that's gross; we have to import React there to use JSX; it's a devDependency of the module.

File: .npmrc [hunk 1]
```diff
@@ -2,3 +2,4 @@ auto-install-peers=true
 public-hoist-pattern[]='*eslint*'
 public-hoist-pattern[]='*prettier*'
 public-hoist-pattern[]='@emotion/react'
+strict-peer-dependencies=false
```
Comment (hasparus) [inline .npmrc]: As you mentioned you'd like to get rid of this from `.npmrc`, you might find useful the `"pnpm": {` settings in package.json. I recently used it to work around dependency problems.

https://github.com/system-ui/theme-ui/blob/27ea4123d548d67891d3a2c1f811b41ea86398d4/package.json#L94-L110

File: jest.config.js [hunk 2]
```diff
@@ -4,14 +4,21 @@
 /** @type {import('@jest/types').Config.InitialOptions} */
 const config = {
   preset: 'ts-jest/presets/js-with-ts',
-  globals: {
-    'ts-jest': {
-      tsconfig: './tsconfig.test.json',
-      useESM: true,
-      diagnostics: {
-        exclude: ['**'],
+  transform: {
```
Comment (hasparus) [inline jest.config.js:7]: This makes so much more sense now.

**Review by hasparus — APPROVED** [review]

File: .npmrc (hunk 1 above)
Comment (lachlanjc) [reply .npmrc]: Oooh thanks for the tip! Even temporarily, felt like a terrible fix 🙈

File: packages/prism/test/index.tsx [hunk 3]
```diff
@@ -2,15 +2,16 @@
  * @jest-environment jsdom
  */
 
-import React, { ReactElement } from 'react'
+import React from 'react'
```
Comment (lachlanjc) [inline packages/prism/test/index.tsx:5]: @hasparus Could you quickly advise on why https://github.com/system-ui/theme-ui/actions/runs/3063545168/jobs/4945735744#step:7:159 is coming up? It's totally fine if I `ts-expect-error` this import but I can't get it to work…

File: .npmrc (hunk 1 above)
Comment (lachlanjc) [reply .npmrc]: Just switched to this.

**Comment by lachlanjc** [comment]
Merging this for now, since it’s been over a month waiting on the Emotion package release. We can easily remove our version override in the future when it’s released, but I don’t want to deal with more merge conflicts down the road.

**Comment by lachlanjc** [comment]
I’d like to merge this for now, since it’s been over a month waiting on the Emotion package release. We can easily remove our version override in the future when it’s released, but I don’t want to deal with more merge conflicts down the road. @hasparus can you take a quick look at the React import error?

**Comment by lachlanjc** [comment]
Ooh makes a ton of sense, thanks!

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.15.3`](https://github.com/system-ui/theme-ui/releases/tag/v0.15.3) :rocket:

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.15.3`](https://github.com/system-ui/theme-ui/releases/tag/v0.15.3) :rocket:

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.15.3`](https://github.com/system-ui/theme-ui/releases/tag/v0.15.3) :rocket:

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.15.3`](https://github.com/system-ui/theme-ui/releases/tag/v0.15.3) :rocket:

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.15.3`](https://github.com/system-ui/theme-ui/releases/tag/v0.15.3) :rocket:

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.15.3`](https://github.com/system-ui/theme-ui/releases/tag/v0.15.3) :rocket:

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.15.4-develop.0`](https://github.com/system-ui/theme-ui/releases/tag/v0.15.4-develop.0) :rocket:

**Pull request merged**